src/goto-instrument/insert_final_assert_false.h

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************\
+
+Module: Insert assert(false) at end of entry function
+
+Author: Nathan Chong, Kareem Khazem
+
+\*******************************************************************/
+
+/// \file
+/// Insert final assert false into a function
+
+/// This transform can be used to check for some cases of vacuous proofs
+/// Usually the given function is the test harness / entry point
+/// The instrumented assert is expected to fail
+/// If it does not then this may indicate inconsistent assumptions
+
+#ifndef CPROVER_GOTO_INSTRUMENT_INSERT_FINAL_ASSERT_FALSE_H
+#define CPROVER_GOTO_INSTRUMENT_INSERT_FINAL_ASSERT_FALSE_H
+
+#include <string>
+
+#include <util/message.h>
+
+class goto_modelt;
+
+class insert_final_assert_falset : public messaget
```

Comment:
I would prefer that this class has a `messaget` as a member, rather than inheriting from it. I would do
```c++
private:
  messaget &log;
```
and the constructor can be:
```c++
insert_final_assert_falset::insert_final_assert_falset(
  message_handlert &_message_handler)
  : log(_message_handler)
```
then you can call `log.error()` instead of just `error()` and so on.

Reply:
Thanks Kareem. There's a note in `CODING_STANDARD.md` that suggests deriving from `messaget`. Which way is preferable?

Reply:
Ah, I remembered being told the opposite by a reviewer at some point. I suppose the CODING_STANDARD.md is the truth. @smowton do you have an opinion on this?

(FWIW I still believe having a messaget member is cleaner, but I'll defer to whatever the prevalent style is)

Reply:
I wouldn't trust the CODING_STANDARD.md as far as that, to be honest. `insert_final_assert_falset` isn't a type of `messaget`, it just wants to use it - so it shouldn't derive from `messaget`. If the coding standard still says that, then it should be updated. :-)

Reply:
I went with messaget as a class member. I'll leave updating the CODING_STANDARD to someone else!